doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,224 @@
+arma::Row<size_t> labels =
+    arma::randi<arma::Row<size_t>>(1000, arma::distr_param(0, 4));
+arma::mat testDataset(10, 500, arma::fill::randu); // 500 test points.
+size_t numWeakLearners = 10;
+size_t numClasses = 5;
+
+mlpack::GradBoosting gb;               // Step 1: create model.
+gb.Train(dataset, labels, numClasses, numWeakLearners);          // Step 2: train model.
+arma::Row<size_t> predictions;
+gb.Classify(testDataset, predictions); // Step 3: classify points.
+
+// Print some information about the test predictions.
+std::cout << arma::accu(predictions == 2) << " test points classified as class "
+    << "2." << std::endl;
+```
+<p style="text-align: center; font-size: 85%"><a href="#simple-examples">More examples...</a></p>
+
+#### Quick links:
+
+ * [Constructors](#constructors): create `GradientBoosting` objects.
+ * [`Train()`](#training): train model.
+ * [`Classify()`](#classification): classify with a trained model.
+ * [Other functionality](#other-functionality) for loading, saving, and
+   inspecting.
+ * [Examples](#simple-examples) of simple usage and links to detailed example
+   projects.
+
+#### See also:
+
+ * [`AdaBoost`](adaboost.md)
+ * [`DecisionTree`](decision_tree.md)
+ * [`DecisionTreeRegressor`](decision_tree_regressor.md)
+ * [Random forests](random_forest.md)
+ * [mlpack classifiers](../../index.md#classification-algorithms)
+ * [Gradient Boosting on Wikipedia](https://en.wikipedia.org/wiki/Gradient_boosting)
+
+### Constructors
+
+ * `gb = GradBoosting()`
+   - Initialize model without training.
+   - You will need to call [`Train()`](#training) later to train the model before
+     calling [`Classify()`](#classification).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners)`
+   - Train on numerical and/or categorical data using default arguments for the weak learners (Decision Trees).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical and/or categorical data, using the given hyperparameters.
+
+---
+
+#### Constructor Parameters:
+
+| **name** | **type** | **description** | **default** |
+|----------|----------|-----------------|-------------|
+| `data` | [`arma::mat`](../matrices.md) | [Column-major](../matrices.md#representing-data-in-mlpack) training matrix. | _(N/A)_ |
+| `datasetInfo` | [`data::DatasetInfo`](https://github.com/mlpack/mlpack/blob/master/doc/user/load_save.md#loading-categorical-data) | Dataset information, specifying type information for each dimension. | _(N/A)_ |
+| `labels` | [`arma::Row<size_t>`](../matrices.md) | Training labels, [between `0` and `numClasses - 1`](../load_save.md#normalizing-labels) (inclusive).  Should have length `data.n_cols`.  | _(N/A)_ |
+| `numClasses` | `size_t` | Number of classes in the dataset. | _(N/A)_ |
+| `numWeakLearners` | `size_t` | Number of weak learner models to be used. | _(N/A)_ |
```

Comment:
I think it would be nice to specify a default argument here; after all, `RandomForest` and `AdaBoost` have defaults.